diff -Naur kdelibs-4.4.5.old/kdeui/dialogs/kshortcutseditor.cpp kdelibs-4.4.5.new/kdeui/dialogs/kshortcutseditor.cpp
--- kdelibs-4.4.5.old/kdeui/dialogs/kshortcutseditor.cpp	2009-10-15 18:36:05.000000000 +0200
+++ kdelibs-4.4.5.new/kdeui/dialogs/kshortcutseditor.cpp	2010-12-20 23:57:03.000000000 +0100
@@ -295,7 +295,7 @@
     q->layout()->setMargin(0);
     ui.searchFilter->searchLine()->setTreeWidget(ui.list); // Plug into search line
     ui.list->header()->setResizeMode(QHeaderView::ResizeToContents);
-    ui.list->header()->hideSection(GlobalAlternate);  //not expected to be very useful
+    //ui.list->header()->hideSection(GlobalAlternate);  //not expected to be very useful
     ui.list->header()->hideSection(ShapeGesture);  //mouse gestures didn't make it in time...
     ui.list->header()->hideSection(RockerGesture);
     if (!(actionTypes & KShortcutsEditor::GlobalAction)) {
