From f5dbca43a4a6842322f2fde4f94120f26880b952 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20Mor=C3=A1vek=20=5BXificurk=5D?= <petr@pada.cz>
Date: Sun, 5 Oct 2014 11:33:34 +0200
Subject: [PATCH] Fix invalid import.

---
 GExiv2.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/GExiv2.py b/GExiv2.py
index 3b257e9..941edc8 100644
--- a/GExiv2.py
+++ b/GExiv2.py
@@ -21,7 +21,6 @@
 from datetime import datetime
 from fractions import Fraction
 
-from gi import _glib
 from gi.repository import GObject
 from ..overrides import override
 from ..importer import modules
-- 
1.8.5.5

