diff -uNr gexiv2.old/gexiv2-metadata.cpp gexiv2/gexiv2-metadata.cpp
--- gexiv2.old/gexiv2-metadata.cpp	2013-03-18 23:16:44.000000000 +0100
+++ gexiv2/gexiv2-metadata.cpp	2013-05-15 19:29:02.705792168 +0200
@@ -561,30 +561,6 @@
     if (str != NULL && *str != '\0')
         return g_strdup(str);
     
-    str = gexiv2_metadata_get_exif_tag_interpreted_string (self, "Exif.Image.ImageDescription");
-    if (str != NULL && *str != '\0')
-        return str;
-    else
-        g_free (str);
-    
-    str = gexiv2_metadata_get_exif_tag_interpreted_string (self, "Exif.Photo.UserComment");
-    if (str != NULL && *str != '\0')
-        return str;
-    else
-        g_free (str);
-    
-    str = gexiv2_metadata_get_exif_tag_interpreted_string (self, "Exif.Image.XPComment");
-    if (str != NULL && *str != '\0')
-        return str;
-    else
-        g_free (str);
-    
-    str = gexiv2_metadata_get_iptc_tag_interpreted_string (self, "Iptc.Application2.Caption");
-    if (str != NULL && *str != '\0')
-        return str;
-    else
-        g_free (str);
-    
     return NULL;
 }
 
@@ -592,14 +568,7 @@
     g_return_if_fail(GEXIV2_IS_METADATA (self));
     g_return_if_fail(self->priv->image.get() != NULL);
     
-    Exiv2::ExifData& exif_data = self->priv->image->exifData();
-    Exiv2::IptcData& iptc_data = self->priv->image->iptcData();
-    
     gexiv2_metadata_set_comment_internal (self, (comment != NULL) ? comment : "");
-    exif_data ["Exif.Image.ImageDescription"] = comment;
-    exif_data ["Exif.Photo.UserComment"] = comment;
-    exif_data ["Exif.Image.XPComment"] = comment;
-    iptc_data ["Iptc.Application2.Caption"] = comment;
 }
 
 void gexiv2_metadata_clear_comment (GExiv2Metadata *self) {
