diff -Naur kdebase-4.4.5.old/apps/konsole/src/ViewContainer.cpp kdebase-4.4.5.new/apps/konsole/src/ViewContainer.cpp
--- kdebase-4.4.5.old/apps/konsole/src/ViewContainer.cpp	2009-12-11 00:13:58.000000000 +0100
+++ kdebase-4.4.5.new/apps/konsole/src/ViewContainer.cpp	2010-12-21 00:39:02.000000000 +0100
@@ -259,6 +259,8 @@
     , _dropIndicatorIndex(-1)
     , _drawIndicatorDisabled(false)
 {
+    setStyleSheet("QTabBar::tab { min-width: 2em; max-width: 25em }");
+    setElideMode(Qt::ElideLeft);
 }
 void ViewContainerTabBar::setDropIndicator(int index, bool drawDisabled)
 {
@@ -394,6 +396,7 @@
         event->ignore();
 }
 
+/* A stylesheet is used now to handle width of tab. */
 QSize ViewContainerTabBar::tabSizeHint(int index) const
 {
      return QTabBar::tabSizeHint(index);
@@ -462,14 +465,12 @@
     _tabBarLayout->addWidget(_newTabButton);
     _tabBarLayout->addWidget(_tabBar);
     _tabBarLayout->addWidget(_closeTabButton); 
-    _tabBarSpacer = new QSpacerItem(0,TabBarSpace);
 
     _layout->addWidget(_stackWidget);
     searchBar()->setParent(_containerWidget);
     if ( position == NavigationPositionTop )
     {
         _layout->insertLayout(0,_tabBarLayout);
-        _layout->insertItemAt(0,_tabBarSpacer);
         _layout->insertWidget(-1,searchBar());
         _tabBar->setShape(QTabBar::RoundedNorth);
     }
@@ -477,7 +478,6 @@
     {
         _layout->insertWidget(-1,searchBar());
         _layout->insertLayout(-1,_tabBarLayout);
-        _layout->insertItemAt(-1,_tabBarSpacer);
         _tabBar->setShape(QTabBar::RoundedSouth);
     }
     else
@@ -520,14 +520,6 @@
     _tabBar->setVisible(visible);
     _newTabButton->setVisible(visible && (features() & QuickNewView));
     _closeTabButton->setVisible(visible && (features() & QuickCloseView));
-    if ( visible )
-    {
-        _tabBarSpacer->changeSize(0,TabBarSpace);
-    }
-    else
-    {
-        _tabBarSpacer->changeSize(0,0);
-    } 
 }
 QList<ViewContainer::NavigationPosition> TabbedViewContainer::supportedNavigationPositions() const
 {
@@ -535,9 +527,9 @@
 }
 void TabbedViewContainer::navigationPositionChanged(NavigationPosition position)
 {
-    // this method assumes that there are only three items 
+    // this method assumes that there are only two items 
     // in the layout
-    Q_ASSERT( _layout->count() == 4 );
+    Q_ASSERT( _layout->count() == 3 );
 
     // index of stack widget in the layout when tab bar is at the bottom
     const int StackIndexWithTabBottom = 0;
@@ -546,11 +538,9 @@
             && _layout->indexOf(_stackWidget) == StackIndexWithTabBottom )
     {
         _layout->removeItem(_tabBarLayout);
-        _layout->removeItem(_tabBarSpacer);
         _layout->removeWidget(searchBar());
 
         _layout->insertLayout(0,_tabBarLayout);
-        _layout->insertItemAt(0,_tabBarSpacer);
         _layout->insertWidget(-1,searchBar());
         _tabBar->setShape(QTabBar::RoundedNorth);
     }
@@ -558,12 +548,10 @@
             && _layout->indexOf(_stackWidget) != StackIndexWithTabBottom )
     {
         _layout->removeItem(_tabBarLayout);
-        _layout->removeItem(_tabBarSpacer);
         _layout->removeWidget(searchBar());
 
         _layout->insertWidget(-1,searchBar());
         _layout->insertLayout(-1,_tabBarLayout);
-        _layout->insertItemAt(-1,_tabBarSpacer);
         _tabBar->setShape(QTabBar::RoundedSouth);
     }
 }
@@ -742,22 +730,11 @@
 
 void TabbedViewContainer::updateTitle(ViewProperties* item)
 {
-    // prevent tab titles from becoming overly-long as this limits the number
-    // of tabs which can fit in the tab bar.  
-    //
-    // if the view's title is overly long then trim it and select the 
-    // right-most 20 characters (assuming they contain the most useful
-    // information) and insert an elide at the front
-    const int MAX_TAB_TEXT_LENGTH = 20;
-
     QListIterator<QWidget*> iter(widgetsForItem(item));
     while ( iter.hasNext() )
     {
         const int index = _stackWidget->indexOf( iter.next() );
-
         QString tabText = item->title();
-        if (tabText.count() > MAX_TAB_TEXT_LENGTH)
-            tabText = tabText.right(MAX_TAB_TEXT_LENGTH).prepend("...");
 
         _tabBar->setTabText( index , tabText );
     }
