diff --git a/src/build.py b/src/build.py
index b4b725b..075bf8c 100644
--- a/src/build.py
+++ b/src/build.py
@@ -37,9 +37,6 @@ def call(cmd, silent=True):
     elif not silent:
         print stderr
 
-def ldconfig(*args,**kwargs):
-    call('ldconfig')
-
 if env['LINKING'] == 'static':
     lib_env.Append(CXXFLAGS="-fPIC")
 
@@ -326,8 +323,6 @@ if env['PLATFORM'] != 'Darwin':
     if 'uninstall' not in COMMAND_LINE_TARGETS:
       result = env.InstallAs(target=target, source=mapnik)
       env.Alias(target='install', source=result)
-      if result:
-            env.AddPostAction(result, ldconfig)
 
     
     # Install symlinks
-- 
1.7.3.4

