diff --git a/SConstruct b/SConstruct
index c2ad084..28f34eb 100644
--- a/SConstruct
+++ b/SConstruct
@@ -344,13 +344,15 @@ opts.AddVariables(
     PathVariable('TIFF_LIBS', 'Search path for libtiff library files', '/usr/' + LIBDIR_SCHEMA, PathVariable.PathAccept),
     PathVariable('PROJ_INCLUDES', 'Search path for PROJ.4 include files', '/usr/local/include', PathVariable.PathAccept),
     PathVariable('PROJ_LIBS', 'Search path for PROJ.4 library files', '/usr/local/' + LIBDIR_SCHEMA, PathVariable.PathAccept),
+    PathVariable('AGG_INCLUDES', 'Search path for AGG include files', '/usr/include/agg2', PathVariable.PathAccept),
+    PathVariable('AGG_LIBS', 'Search path for AGG library files', '/usr/' + LIBDIR_SCHEMA, PathVariable.PathAccept),
     ('PKG_CONFIG_PATH', 'Use this path to point pkg-config to .pc files instead of the PKG_CONFIG_PATH environment setting',''),
     
     # Variables affecting rendering back-ends
     
     BoolVariable('RENDERING_STATS', 'Output rendering statistics during style processing', 'False'),
     
-    BoolVariable('INTERNAL_LIBAGG', 'Use provided libagg', 'True'),
+    BoolVariable('INTERNAL_LIBAGG', 'Use provided libagg', 'False'),
 
     BoolVariable('SVG_RENDERER', 'build support for native svg renderer', 'False'),
     
@@ -1098,6 +1100,7 @@ if not preconfigured:
         ['z', 'zlib.h', True,'C'],
         ['proj', 'proj_api.h', True,'C'],
         [env['ICU_LIB_NAME'],'unicode/unistr.h',True,'C++'],
+        ['agg','agg2/agg_config.h', True,'C++'],
     ]
 
     if env['JPEG']:
diff --git a/src/build.py b/src/build.py
index c3227be..b4b725b 100644
--- a/src/build.py
+++ b/src/build.py
@@ -55,7 +55,7 @@ filesystem = 'boost_filesystem%s' % env['BOOST_APPEND']
 regex = 'boost_regex%s' % env['BOOST_APPEND']
 
 # clear out and re-set libs for this env
-lib_env['LIBS'] = ['freetype','ltdl','png','tiff','z','jpeg','proj',env['ICU_LIB_NAME'],filesystem,regex]
+lib_env['LIBS'] = ['freetype','ltdl','png','tiff','z','jpeg','proj',env['ICU_LIB_NAME'],'agg',filesystem,regex]
 
 if len(env['EXTRA_FREETYPE_LIBS']):
     lib_env['LIBS'].extend(copy(env['EXTRA_FREETYPE_LIBS']))
-- 
1.7.3.4

