--- src/SConscript	2011-08-17 10:40:02.049638844 +0200
+++ src/SConscript	2011-08-17 10:40:02.050638835 +0200
@@ -37,9 +37,6 @@
     elif not silent:
         print stderr
 
-def ldconfig(*args,**kwargs):
-    call('ldconfig')
-
 
 if env['LINKING'] == 'static':
     lib_env.Append(CXXFLAGS="-fPIC")
@@ -309,8 +306,6 @@
     if 'uninstall' not in COMMAND_LINE_TARGETS:
       result = env.InstallAs(target=target, source=mapnik)
       env.Alias(target='install', source=result)
-      if result:
-            env.AddPostAction(result, ldconfig)
 
     
     # Install symlinks
