diff -urN opencachemanager-0.21.old/ocm-gtk/Main.cs opencachemanager-0.21/ocm-gtk/Main.cs
--- opencachemanager-0.21.old/ocm-gtk/Main.cs	2010-10-04 01:06:19.000000000 +0200
+++ opencachemanager-0.21/ocm-gtk/Main.cs	2010-10-13 09:51:08.000000000 +0200
@@ -60,7 +60,7 @@
 					m_file = args[0];
 			//System.Console.WriteLine("Path is " + "/usr/local/share/locale");
 			//Mono.Unix.Catalog.Init ("opencachemanager", "./locale");
-			Mono.Unix.Catalog.Init ("opencachemanager", "/usr/local/share/locale");
+			Mono.Unix.Catalog.Init ("opencachemanager", "/usr/share/locale");
 			bool runWizard = false;
 			try {
 				GConf.Client client = new GConf.Client ();
diff -urN opencachemanager-0.21.old/ocm-gtk/ocm-gtk.in opencachemanager-0.21/ocm-gtk/ocm-gtk.in
--- opencachemanager-0.21.old/ocm-gtk/ocm-gtk.in	2010-10-04 01:06:09.000000000 +0200
+++ opencachemanager-0.21/ocm-gtk/ocm-gtk.in	2010-10-13 09:53:40.000000000 +0200
@@ -1,3 +1,3 @@
 #!/bin/sh
-cd "/usr/local/lib/opencachemanager"
-exec mono "/usr/local/lib/opencachemanager/ocm-gtk.exe" "$@"
+cd "/usr/lib/opencachemanager"
+exec mono "/usr/lib/opencachemanager/ocm-gtk.exe" "$@"
